### PR TITLE
React to project.json schema changes and tooling package renames

### PR DESCRIPTION
**Pull request by phenning**
//cc @DamianEdwards @rustd @anurse @moozzyk @prafullbhosale @NTaylorMullen

### CODE REVIEW

File: src/BaseTemplates/EmptyWeb/project.json [hunk 1]
```diff
@@ -26,25 +26,22 @@
...
-
-  "exclude": [
-    "node_modules"
-  ],
+  "publishOptions": {
+    "include": [
+      "wwwroot",
+      "web.config"
+    ],
+    "exclude": [ "node_modules" ]
```
Comment (DamianEdwards) [inline src/BaseTemplates/EmptyWeb/project.json]: The `"exclude"` shouldn't be needed at all. You will however need to include the `appsettings.json`

Comment (DamianEdwards) [inline src/BaseTemplates/EmptyWeb/project.json]: And you need to include "Views" for MVC apps too

Comment (DamianEdwards) [inline src/BaseTemplates/EmptyWeb/project.json]: Just saw these are separate files, carry on.

File: src/BaseTemplates/StarterWeb/project.json [hunk 2]
```diff
@@ -43,27 +43,24 @@
...
-    "node_modules"
-  ],
+  "publishOptions": {
+    "include": [
+      "wwwroot",
+      "Views",
+      "appsettings.json",
+      "web.config"
+    ],
+    "exclude": [ "node_modules" ]
```
Comment (DamianEdwards) [inline src/BaseTemplates/StarterWeb/project.json]: Shouldn't need this `"exclude"` as above

File: src/BaseTemplates/WebAPI/project.json [hunk 3]
```diff
@@ -33,27 +33,24 @@
...
-    "node_modules"
-  ],
+  "publishOptions": {
+    "include": [
+      "wwwroot",
+      "Views",
+      "appsettings.json",
+      "web.config"
+    ],
+    "exclude": [ "node_modules" ]
```
Comment (DamianEdwards) [inline src/BaseTemplates/WebAPI/project.json]: Shouldn't need this

File: src/Rules/StarterWeb/AI/IndividualAuth/project.json [hunk 4]
```diff
@@ -61,6 +58,13 @@
         "portable-net45+win8"
       ]
     },
+    "Microsoft.Extensions.CodeGeneration.Tools": {
+      "version": "__AspNetCorePlatformVersion__",
+      "imports": [
+        "portable-net45+win8+dnxcore50",
```
Comment (DamianEdwards) [inline src/Rules/StarterWeb/AI/IndividualAuth/project.json:64]: Do we know why **both** of these imports are required here?

File: src/Rules/StarterWeb/AI/IndividualAuth/project.json [hunk 5]
```diff
@@ -78,27 +82,24 @@
...
-    "node_modules"
-  ],
+  "publishOptions": {
+    "include": [
+      "wwwroot",
+      "Views",
+      "appsettings.json",
+      "web.config"
+    ],
+    "exclude": [ "node_modules" ]
```
Comment (DamianEdwards) [inline src/Rules/StarterWeb/AI/IndividualAuth/project.json]: Shouldn't need this

File: src/Rules/StarterWeb/AI/NoAuth/project.json [hunk 6]
```diff
@@ -44,27 +44,24 @@
...
-    "node_modules"
-  ],
+  "publishOptions": {
+    "include": [
+      "wwwroot",
+      "Views",
+      "appsettings.json",
+      "web.config"
+    ],
+    "exclude": [ "node_modules" ]
```
Comment (DamianEdwards) [inline src/Rules/StarterWeb/AI/NoAuth/project.json]: Shouldn't need this

File: src/BaseTemplates/WebAPI/project.json (hunk 3 above)
Comment (phenning) [inline src/BaseTemplates/WebAPI/project.json]: I think the reason we had the exludes in there for all the web templates is so if customers start installing node packages we automatically exclude them, even though these templates don't ship with node modules by default.

File: src/Rules/StarterWeb/AI/IndividualAuth/project.json (hunk 4 above)
Comment (phenning) [inline src/Rules/StarterWeb/AI/IndividualAuth/project.json:64]: Yes, net45+win8+dnxcore50 is required to get the proper version of json.net, just specifying net45+win8 nets you a json.net that has a fx dependency on System.Runtime 2.0.5.0

net45+win8 is required for Ix.Async and Remotion

File: src/Rules/StarterWeb/AI/IndividualAuth/project.json (hunk 5 above)
Comment (phenning) [inline src/Rules/StarterWeb/AI/IndividualAuth/project.json]: Don't we actually need it for all StarterWebs (assuming that its purpose is what I think it is)

**Comment by NTaylorMullen** [comment]
Razor tooling bits look good. :+1:

**Comment by prafullbhosale** [comment]
Scaffolding bits look good too!

File: src/BaseTemplates/WebAPI/project.json (hunk 3 above)
Comment (rustd) [inline src/BaseTemplates/WebAPI/project.json]: yeah. that's what I remember @madskristensen do we need to exclude the node_modules?

Comment (madskristensen) [inline src/BaseTemplates/WebAPI/project.json]: If the template doesn't have a node_modules folder then there is no reason to exclude it at all. That would just be confusing

Comment (rustd) [inline src/BaseTemplates/WebAPI/project.json]: Of course. @phenning we only need it only for the Starter Web templates then which have a node_modules folder

Comment (phenning) [inline src/BaseTemplates/WebAPI/project.json]: We don't need it at all anymore, due to the way concerns will split between publish and build. I confirmed this offline with Damian.

**Pull request merged**